protocol/streaming/full_node_streaming_manager.go

```diff
@@ -220,68 +220,59 @@ func (sm *FullNodeStreamingManagerImpl) getNextAvailableSubscriptionId() uint32
+	orderBookUpdate *clobtypes.StreamUpdate_OrderbookUpdate,
+	subaccountIdNumbers []uint32,
+	logger log.Logger,
+) bool {
+	for _, orderBookUpdate := range orderBookUpdate.OrderbookUpdate.Updates {
+		orderBookUpdateSubaccountIdNumber, err := streaming_util.GetOffChainUpdateV1SubaccountIdNumber(orderBookUpdate)
+		if err == nil {
+			if slices.Contains(subaccountIdNumbers, orderBookUpdateSubaccountIdNumber) {
+				return true
+			}
+		} else {
+			logger.Error(err.Error())
+		}
+	}
+	return false
+}
+
 // Filter StreamUpdates for subaccountIdNumbers
 // If a StreamUpdate_OrderUpdate contains no updates for subscribed subaccounts, drop message
 // If a StreamUpdate_OrderUpdate contains updates for subscribed subaccounts, construct a new
 // StreamUpdate_OrderUpdate with updates only for subscribed subaccounts
-func (sub *OrderbookSubscription) FilterSubaccountStreamUpdates(
-	output chan []clobtypes.StreamUpdate,
+func FilterSubaccountStreamUpdates(
+	updates []clobtypes.StreamUpdate,
+	subaccountIdNumbers []uint32,
 	logger log.Logger,
-) {
-	subaccountIdNumbers := make([]uint32, len(sub.subaccountIds))
-	for i, subaccountId := range sub.subaccountIds {
-		subaccountIdNumbers[i] = subaccountId.Number
-	}
-
+) *[]clobtypes.StreamUpdate {
 	// If reflection becomes too expensive, split updatesChannel by message type
-	for updates := range sub.updatesChannel {
-		filteredUpdates := []clobtypes.StreamUpdate{}
-		for _, update := range updates {
-			switch updateMessage := update.UpdateMessage.(type) {
-			case *clobtypes.StreamUpdate_OrderbookUpdate:
-				orderBookUpdates := []ocutypes.OffChainUpdateV1{}
-				for _, orderBookUpdate := range updateMessage.OrderbookUpdate.Updates {
-					orderBookUpdateSubaccountIdNumber, err := streaming_util.GetOffChainUpdateV1SubaccountIdNumber(orderBookUpdate)
-					if err == nil {
-						if slices.Contains(subaccountIdNumbers, orderBookUpdateSubaccountIdNumber) {
-							orderBookUpdates = append(orderBookUpdates, orderBookUpdate)
-						}
-					} else {
-						logger.Error(err.Error())
-					}
-				}
-				// Drop the StreamUpdate_OrderbookUpdate if all updates inside were dropped
-				if len(orderBookUpdates) > 0 {
-					if len(orderBookUpdates) < len(updateMessage.OrderbookUpdate.Updates) {
-						update = clobtypes.StreamUpdate{
-							BlockHeight: update.BlockHeight,
-							ExecMode:    update.ExecMode,
-							UpdateMessage: &clobtypes.StreamUpdate_OrderbookUpdate{
-								OrderbookUpdate: &clobtypes.StreamOrderbookUpdate{
-									Snapshot: updateMessage.OrderbookUpdate.Snapshot,
-									Updates:  orderBookUpdates,
-								},
-							},
-						}
-					}
-					filteredUpdates = append(filteredUpdates, update)
-				}
-			default:
+	filteredUpdates := []clobtypes.StreamUpdate{}
+	for _, update := range updates {
+		switch updateMessage := update.UpdateMessage.(type) {
+		case *clobtypes.StreamUpdate_OrderbookUpdate:
+			if doFilterSubaccountStreamUpdate(updateMessage, subaccountIdNumbers, logger) {
```

Comment:
nit: rename to `filterStreamUpdateBySubaccount` to disambiguate from "filtering SubaccountStreamUpdate"